### PR TITLE
USHIFT-1061: Move hack/config-gen into cmd/generate-config directory

### PR DESCRIPTION
**Pull request by ggiguash**
Automatic configuration generation and verification is part of the development and CI procedures now. 
Moving the sources and scripts out of `hack` directory.

### CODE REVIEW

File: generate-config/cmd.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ func newCommand() *cobra.Command {
 	opt := configGenOpts{}
 
 	cmd := &cobra.Command{
-		Use:   "config-gen",
+		Use:   "generate-config",
```
Comment (pmtk) [inline generate-config/cmd.go:20]: Could we use `cmd/generate-config/`? It more consistent with go dir pattern and doesn't pollute root dir

Comment (ggiguash) [reply generate-config/cmd.go]: Done

**Comment by ggiguash** [comment]
/retitle USHIFT-1061: Move hack/config-gen into cmd/generate-config directory

**Pull request merged**